torchvision/models/maxvit.py

```diff
@@ -0,0 +1,854 @@
+
+        self.layers = nn.ModuleList()
+        # account for the first stride of the first layer
+        self.grid_size = _get_conv_output_shape(input_grid_size, kernel_size=3, stride=2, padding=1)
+
+        for idx, p in enumerate(p_stochastic):
+            stride = 2 if idx == 0 else 1
+            self.layers += [
+                MaxVitLayer(
+                    in_channels=in_channels if idx == 0 else out_channels,
+                    out_channels=out_channels,
+                    squeeze_ratio=squeeze_ratio,
+                    expansion_ratio=expansion_ratio,
+                    stride=stride,
+                    normalization_fn=normalization_fn,
+                    activation_fn=activation_fn,
+                    head_dim=head_dim,
+                    mlp_ratio=mlp_ratio,
+                    mlp_dropout=mlp_dropout,
+                    attn_dropout=attn_dropout,
+                    partition_size=partition_size,
+                    grid_size=self.grid_size,
+                    p_stochastic_dropout=p,
+                ),
+            ]
+
+    def forward(self, x: Tensor) -> Tensor:
+        """
+        Args:
+            x (Tensor): Input tensor of shape (B, C, H, W).
+        Returns:
+            Tensor: Output tensor of shape (B, C, H, W).
+        """
+        for layer in self.layers:
+            x = layer(x)
+        return x
+
+
+class MaxVit(nn.Module):
+    """
+    Implements MaxVit Transformer from the `MaxViT: Multi-Axis Vision Transformer <https://arxiv.org/abs/2204.01697>`_ paper.
+    Args:
+        input_size (Tuple[int, int]): Size of the input image.
+        input_channels (int): Number of input channels.
+        stem_channels (int): Number of channels in the stem.
+        num_classes (int): Number of classes.
+        block_channels (List[int]): Number of channels in each block.
+        block_layers (List[int]): Number of layers in each block.
+        stochastic_depth_prob (float): Probability of stochastic depth. Expands to a list of probabilities for each layer that scales linearly to the specified value.
+        squeeze_ratio (float): Squeeze ratio in the SE Layer.
+        expansion_ratio (float): Expansion ratio in the MBConv bottleneck.
+        normalization_fn (Callable[..., nn.Module]): Normalization function.
+        activation_fn (Callable[..., nn.Module]): Activation function.
+        head_dim (int): Dimension of the attention heads.
+        mlp_ratio (int): Expansion ratio of the MLP layer.
+        mlp_dropout (float): Dropout probability for the MLP layer.
+        attn_dropout (float): Dropout probability for the attention layer.
+        partition_size (int): Size of the partitions.
+    """
+
+    def __init__(
+        self,
+        # input size parameters
+        input_size: Tuple[int, int],
+        # stem and task parameters
+        input_channels: int,
```

Comment:
No other model at the moment accepts a `input_channels` parameter. It has been a request from the community to introduce it to support single-channel (and other) images. Perhaps it's something we want to do, but I would see it as a coordinated effort to make it available to all models.